### ISSUE TITLE
func azure functionapp publish <FunctionName> doesn't work

### ISSUE THREAD

**Issue by dilantha111** [Azure/azure-functions-core-tools#308]
when issuing the above command with azure-functions-cli, it only returns 

>  Publish C:\Users\dilanthaw\Downloads\FunctionAppDilantha contents to an Azure Function App. Locally deleted files are no
> t removed from destination.
> Getting site publishing info...
> Sequence contains no elements 

And nothing happens. I tried login to the azure with func azure login command. But result is still the same. And just in case I tried the command "func azure functionapp list" as well. But it also gives me the result 
> Sequence contains no elements 
Any help would be appreciated.

**Comment by cmatskas** [comment]
Hi @dilantha111 . Do you have an Azure Function App already deployed on Azure? The Core tools deploy into an existing Function App so if one doesn't exist, the deployment fails. Based on your comments, you haven't deployed one yet so the Core tools fail. Think of the FunctionApp on Azure as your Azure Functions "container" where your code is deployed to run. Please give this a go and let us know if this fixes the problem

**Comment by dilantha111** [comment]
Hi @cmatskas  thanks for the quick response. Yes I have an Azure Function App already running on Azure. And as a workaround I tried deploying via setting up a local git repo. And that works fine. But still this fails.

**Comment by pengo98** [comment]
@dilantha111 I'm experiencing the same issue using exactly the same command for deploying the binaries. I already created an empty Function App in the portal.  Every time I run it, I get `Sequence contains no elements`.

**Comment by dilantha111** [comment]
Hi @pengo98, I think you better take the approach using git. It worked fine rather than publish command

**Comment by jackfoxy** [comment]
using core tools 2.0.1-beta.22 and I am also having this problem
the Function App exists, but it is not empty
I then tried creating an empty function app, and same result

**Comment by jackfoxy** [comment]
This is a problem with the beta. Installing 1.07 works.

**Comment by wk-j** [comment]
Change runtime version to beta its working for me

https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions#target-the-version-20-runtime

**Comment by jackfoxy** [comment]
@wk-j  that is not the problem. It is getting no further than

Response status code does not indicate success: 401 (Unauthorized).

The function app I am targeting is already 2.0 beta.  I already logged in, login-azurermaccount

**Comment by liambarnett** [comment]
I'm still encountering this issue. Fixed by downgrading using 'npm i -g azure-functions-core-tools@1.0.7'

**Comment by lnfnunes** [comment]
Trying to `publish` on the same folder where I run `func host start` to test locally.

![image](https://user-images.githubusercontent.com/2450417/40567710-ebca8244-604c-11e8-9c70-7be07026c8c3.png)
![image](https://user-images.githubusercontent.com/2450417/40567715-f2bea01c-604c-11e8-8711-c678f816ed95.png)

I don't know how to debug this error!
Mac OS High Sierra

**Comment by luisrudge** [comment]
@ahmelsayed I'm still having this issue.

```
➜  snq-audio-to-json-function func -v
2.0.1-beta.35
```

**Comment by delormej** [comment]
Same issue for me; Function App exist in Azure, I get an error "Sequence contains no elements" when running func azure function app publish.  

Function runs fine locally.  Targetting dontetcore-2.1 (C# library), running from Ubuntu machine.  func version is 2.0.1-beta.31

**Comment by ahmelsayed** [comment]
I may have misunderstood the issue here. Can you run `CLI_DEBUG=1 func azure functionapp publish ....` and share the stack trace?

**Comment by pnda489** [comment]
I got the same problem.

CLI_DEBUG=1 func azure functionapp publish dev-service-functions
Publish /Users/mhoefer/GitHub/DSO/Functions/src/TenantDeployment contents to an Azure Function App. Locally deleted files are not removed from destination.
Getting site publishing info...
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Azure.Functions.Cli.Arm.ArmManager.<GetCurrentSubscriptionAsync>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Arm.ArmManager.<GetFunctionAppsAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Arm.ArmManager.<GetFunctionAppAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionApp.<RunAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.ConsoleApp.<RunAsync>d__5`1.MoveNext()

**Comment by pnda489** [comment]
CLI_DEBUG=1 func azure functionapp publish myservicename
Publish /Users/../ contents to an Azure Function App. Locally deleted files are not removed from destination.
Getting site publishing info...
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Azure.Functions.Cli.Arm.ArmManager.<GetCurrentSubscriptionAsync>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Arm.ArmManager.<GetFunctionAppsAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Arm.ArmManager.<GetFunctionAppAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionApp.<RunAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Functions.Cli.ConsoleApp.<RunAsync>d__5`1.MoveNext()


The Function App is deployed in azure, I also tried to use a non Microsoft-Live Account. Tried on Mac and Pc
